pkg/ruler/manager.go

```diff
@@ -0,0 +1,231 @@
+package ruler
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	ot "github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/notifier"
+	promRules "github.com/prometheus/prometheus/rules"
+	"github.com/weaveworks/common/user"
+	"golang.org/x/net/context/ctxhttp"
+
+	store "github.com/cortexproject/cortex/pkg/ruler/rules"
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+var (
+	managersTotal = promauto.NewGauge(prometheus.GaugeOpts{
```

Comment:
We try to avoid global metrics. Can you move it to `ManagerMetrics`?

Reply:
If I understand correctly, `ManagerMetrics` is actually something passed into the Prometheus manager and does not contain ruler level metrics. This particular `managersTotal` metric was already a global metric in `ruler.go` here - https://github.com/cortexproject/cortex/blob/e8a66862fa0500a65ceff575f96616460262c8db/pkg/ruler/ruler.go#L56-L60

Please do correct me if I'm wrong. :)

Reply:
Right. Then you can just define the metric in `DefaultMultiTenantManager`.